References: bsc#1194323
Windows server 2k16, 2k19, and 2k22 have the volumen ID
so libosinfo can't really tell them apart.
This hack after detecting a windows ISO just looks at the
ISO filename for an extra clue.

--- virt-manager-3.2.0/virtinst/install/installer.py.orig	2022-01-20 15:40:27.849623677 -0700
+++ virt-manager-3.2.0/virtinst/install/installer.py	2022-01-20 15:42:22.229628567 -0700
@@ -541,6 +541,14 @@ class Installer(object):
                 osguess = OSDB.guess_os_by_iso(self.cdrom)
                 if osguess:
                     ret = osguess[0]
+                    # Hack because windows ISOs contain same volume ID
+                    if ret and ret.startswith("win"):
+                        if "windows_server_2022" in self.cdrom:
+                            ret = "win2k22"
+                        elif "windows_server_2019" in self.cdrom:
+                            ret = "win2k19"
+                        elif "windows_server_2016" in self.cdrom:
+                            ret = "win2k16"
         else:
             log.debug("No media for distro detection.")
 
